File: api/src/main/java/org/openmrs/module/openhmis/inventory/api/model/Consumption.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package org.openmrs.module.openhmis.inventory.api.model;

import java.util.Date;
import org.codehaus.jackson.annotate.JsonIgnore;
import org.openmrs.module.openhmis.commons.api.entity.model.BaseSerializableOpenmrsMetadata;
import org.openmrs.module.openhmis.commons.api.entity.model.BaseSimpleCustomizableMetadata;
import org.openmrs.module.openhmis.commons.api.entity.model.ISimpleCustomizable;

/**
 * @author MORRISON.I
 */
public class Consumption extends BaseSerializableOpenmrsMetadata {

	public static final long serialVersionUID = 1L;

	private Integer consumptionId;
	private Department department;
	private Item item;
	private Date consumptionDate;
	private Integer quantity;
	private Integer wastage;
	private String batchNumber;

	public String getBatchNumber() {
		return batchNumber;
	}

	public void setBatchNumber(String batchNumber) {
		this.batchNumber = batchNumber;
	}

	public Consumption(Integer consumptionId) {
		this.consumptionId = consumptionId;
	}

	public Consumption() {}

	//private String testType;
	public Integer getWastage() {
		return wastage;
	}

	public void setWastage(Integer wastage) {
		this.wastage = wastage;
	}

	public Department getDepartment() {
		return department;
	}

	public void setDepartment(Department department) {
		this.department = department;
	}

	public Item getItem() {
		return item;
	}

	public void setItem(Item item) {
		this.item = item;
	}

	public Date getConsumptionDate() {
		return consumptionDate;
	}

	public void setConsumptionDate(Date consumptionDate) {
		this.consumptionDate = consumptionDate;
	}

	public Integer getQuantity() {
		return quantity;
	}

	public void setQuantity(Integer quantity) {
		this.quantity = quantity;
	}

	@Override
	public Integer getId() {
		return this.consumptionId;
	}

	@Override
	public void setId(Integer intgr) {
		this.consumptionId = intgr;
	}

	@Override
	@JsonIgnore
	public Boolean getRetired() {
		return super.getRetired();
	}

}

File: omod_1.x/src/main/webapp/resources/js/model/consumption.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
define(
	[
				openhmis.url.backboneBase + 'js/lib/underscore',
		openhmis.url.backboneBase + 'js/model/generic',
		openhmis.url.backboneBase + 'js/lib/i18n',
        openhmis.url.backboneBase + 'js/model/openhmis',
		openhmis.url.inventoryBase + 'js/model/department',
                openhmis.url.inventoryBase + 'js/model/item'
	],
	function(_,openhmis, __) {
		openhmis.Consumption = openhmis.GenericModel.extend({
			meta: {
				name: __(openhmis.getMessage('openhmis.inventory.consumption.name')),
				namePlural: __(openhmis.getMessage('openhmis.inventory.consumption.namePlural')),
				openmrsType: 'metadata',
				restUrl: openhmis.url.inventoryModelBase + 'consumption'
			},

			schema: {
				department: {
					type: 'DepartmentSelect',
					options: new openhmis.GenericCollection(null, {
						model: openhmis.Department,
						url: openhmis.url.inventoryModelBase + 'department'
					}),
					objRef: true
				},
                                item: {
					type: 'ItemSelect',
					options: new openhmis.GenericCollection(null, {
						model: openhmis.Item,
						url: openhmis.url.inventoryModelBase + 'item'
					}),
					objRef: true
				},
                                quantity: { type: "BasicNumber" },
                                consumptionDate: {
                    type: 'OperationDateEditor',
                    format: openhmis.dateTimeFormatLocale
                },
                wastage:{type : "BasicNumber"}
                
			}

		
		});

		return openhmis;
	}
);


